section_faiss_vector_index.go

```diff
@@ -564,32 +564,41 @@ func (vo *vectorIndexOpaque) process(field index.VectorField, fieldID uint16, do
 	dim := field.Dims()
 	metric := field.Similarity()
 
-	if vec != nil {
+	// Embedder is supposed to make sure len(vec) is a multiple of dim.
+	// Not double checking it here to avoid the overhead.
+	numSubVecs := len(vec) / int(dim)
+	for i := 0; i < numSubVecs; i++ {
+		subVec := vec[i*int(dim) : (i+1)*int(dim)]
+
+		if subVec == nil {
```

Comment:
When can this be `nil`? And why continue if it is?

Reply:
done